core/include/detray/propagator/actors/aborters.hpp

```diff
@@ -93,4 +93,28 @@ struct target_aborter : actor {
     }
 };
 
+/// Aborter triggered when the next surface is reached
+struct next_surface_aborter : actor {
+    struct state {
+        // minimal step length to prevent from staying on the same surface
+        scalar min_step_length = 0.;
+        bool success = false;
+    };
+
+    template <typename propagator_state_t>
+    DETRAY_HOST_DEVICE void operator()(state &abrt_state,
+                                       propagator_state_t &prop_state) const {
+
+        auto &navigation = prop_state._navigation;
+        auto &stepping = prop_state._stepping;
+
+        // Abort at the next sensitive surface
+        if (navigation.is_on_sensitive() &&
+            stepping._s > abrt_state.min_step_length) {
+            navigation.exit();
```

Comment:
This is not going to work, I believe. It should be ```prop_state._heartbeat &= navigation.exit();```

Reply:
OK. BTW what is `_heartbeat` of navigator for?

Reply:
I think some things were just easier that way, I don't really remember tbh
https://github.com/acts-project/detray/blob/6a9c3c9e1bef818687955b7af8f2955bcfea596f/core/include/detray/propagator/navigator.hpp#L293

Reply:
ok